[TorchScript, PT2] Add torch._check compatibility support

gmagogsfm opened the pull request:
Summary:
Add support for torch._check() in TorchScript jit.script frontend. 

* It will be special cased to behave like torch._assert, turned into an if + raise exception.

Test Plan:
Unit tests

Rollback Plan:

Differential Revision: D79744604




cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel

davidberard98 reviewed (approved):
Generally LGTM, thanks!

Minor nit - arg processing logic in sugared_value.cpp seems a bit unnecessarily complicated / missing a case, but otherwise everything looks good!

gmagogsfm committed:
[TorchScript, PT2] Add torch._check compatibility support (#159988)

Summary:
Add support for torch._check() in TorchScript jit.script frontend.

* It will be special cased to behave like torch._assert, turned into an if + raise exception.


Test Plan:
Unit tests

Rollback Plan: cc EikanWang jgong5 wenzhe-nrv sanchitintel

Reviewed By: davidberard98

Differential Revision: D79744604

Pulled By: gmagogsfm
